### DISCUSSION TITLE
Buffer locking, view/read-only buffers

### DISCUSSION THREAD

**Discussion by zambetti** [zed-industries/zed#9218] [Feature Requests]
After having migrated to Zed as a daily driver, one feature I'm missing from BBEdit is the ability to 'lock' a document ... that is, having it open, but as view/read-only. This is also an inherent behavior of Vi with normal vs insert mode; it's not called 'locking' per se, but it is equivalent to files opening in a 'locked' state.

With many files open at once, I'm usually only editing one or two, the rest are for reference only; so the lock gives peace of mind that no changes are made by accident.

If this were supported, one way to accommodate it in the UI is to have a lock/unlock icon next to the buffer search magnifying glass (since it is a buffer feature), along with a setting like "buffer_locking": { "button": false, "lock_on_open": false }

Is such a lock feature widely appealing? Or is it only comforting for those paranoid few?

Mockup of how it might look...

![lock](https://github.com/zed-industries/zed/assets/98515/0c1b0831-7449-4ec5-a3c6-bd58b09098a9)

Screenshot of the BBEdit lock...

![Screenshot 2024-03-12 at 12 57 44](https://github.com/zed-industries/zed/assets/98515/eb1712ea-3bd1-4bfa-8aff-3723e01d01db)

**Comment by Moshyfawn** [comment]
I believe that while this can come in handy for extra peace of mind, this feature, or rather the implementation of it, can tie in nicely with other use cases, such as marking certain buffers as read-only, as suggested in #5105.

**Reply by raveenb** [reply]
yes, this can be very useful, since zed starts up almost instantaneously. perhaps a command line arg can be given

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
